book.md

```diff
@@ -109,7 +109,73 @@ feedback to be received through <https://github.com/llsoftsec/llsoftsecbook>.
  
 # Memory vulnerability based attacks and mitigations
 
-\missingcontent{Write chapter on memory vulnerabilities and mitigation.}
+## Introduction
+
+An important class of vulnerabilities arise due to memory access errors, such as
+buffer overflows, use-after-free accesses and null pointer dereferences. An
+attacker can exploit such vulnerabilities to leak sensitive data or
+overwrite critical memory locations and gain control of the vulnerable program.
+These types of vulnerabilities most commonly occur due to programming errors when
+using languages such as C/C++, which do not provide mechanisms to protect the
+integrity of memory accesses by default.
+
+Memory vulnerabilities have a long history. The Morris worm in 1988 was the
+first widely publicized attack exploiting a buffer overflow. Later, in the
+mid-90s a few famous write-ups describing buffer overflows appeared
+[@AlephOne1996].  Stack overflows were mitigated with stack canaries and
+non-executable stacks, but attackers answered with more ingenious ways to
+bypass these mitigations: code reuse attacks, starting from simple attacks like
+return-into-libc [@Solar1997] and leading up to Return-Oriented Programming
+[@Shacham2007] and even more complex techniques.
+
+To defend against code reuse attacks, the defensive security side came up with
+address space layout randomization (ASLR) and control-flow integrity (CFI)
+measures. The race continues to this day. Each newly deployed mitigation
+results in attempts, often successful, to bypass it, or in alternative, more
+complex exploitation techniques, and even tools to automate them.
+
+Memory safe [@Hicks2014] languages are designed with prevention of such
+vulnerabilities in mind and use techniques such as bounds checking and
+automatic memory management. If these languages promise to eliminate
+memory vulnerabilities, why are we still discussing this topic?
+
+On one hand, C and C++ remain very popular languages, particular in the
```

Comment:
Minor minor style question: "On one hand" versus "On the one hand"?
I found this https://jakubmarian.com/on-the-one-hand-on-one-hand/ suggesting "On the one hand" might be preferred. Maybe best to check if there's anything about this in CMoS?

Reply:
I didn't find specific guidance in CMoS, but there are uses of "on the one hand", so I'll use this version too.